src/libraries/System.Private.CoreLib/src/System/Globalization/TimeSpanParse.cs

```diff
@@ -1440,11 +1437,12 @@ private static bool ParseExactDigits(ref TimeSpanTokenizer tokenizer, int minDig
             return tokenLength >= minDigitLength;
         }
 
-        private static bool ParseExactLiteral(ref TimeSpanTokenizer tokenizer, StringBuilder enquotedString)
+        private static bool ParseExactLiteral(ref TimeSpanTokenizer tokenizer, ref ValueStringBuilder enquotedString)
         {
-            for (int i = 0; i < enquotedString.Length; i++)
+            ReadOnlySpan<char> span = enquotedString.AsSpan();
+            for (int i = 0; i < span.Length; i++)
             {
-                if (enquotedString[i] != tokenizer.NextChar)
+                if (span[i] != tokenizer.NextChar())
```

Comment:
what is the value of the Span here -- is indexing into it faster than indexing into a string?

Reply:
It wasn't a string. It was a StringBuilder, and yes, indexing into a span is way faster than indexing into a StringBuilder.

Reply:
I of all people should know that: https://github.com/dotnet/runtime/issues/24395. Perhaps I should open an analyzers suggestion.